hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2613,6 +2615,73 @@ public ServiceInfoEx getServiceInfo() throws IOException {
     return new ServiceInfoEx(getServiceList(), caCertPem);
   }
 
+  private final List<String> finalizationMsgs = new ArrayList<>();
+  private UpgradeFinalizationStatus.Status finalizationStatus =
+      UpgradeFinalizationStatus.Status.FINALIZATION_REQUIRED;
+
+  @Override
+  public UpgradeFinalizationStatus finalizeUpgrade(String upgradeClientID)
+      throws IOException {
+    if (!finalizationStatus
+        .equals(UpgradeFinalizationStatus.Status.FINALIZATION_REQUIRED)){
+      throw new OMException("Finalization is not needed.", INVALID_REQUEST);
```

Comment:
As I commented, FINALIZATION_REQUIRED and INVALID_REQUEST could use a better more understandable name, : ).

Reply:
As written to the previous comments, this part of the code will be changed with the server side logic implemented where we are still experimenting with a few things. But yes error messages has to be better, though states might remain as they are just handled fully as described in the comment for the CientProtocol.proto

Reply:
Comments makes sense to me.